packages/react-composites/src/composites/ChatComposite/file-sharing/UploadedFile.tsx

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { EventEmitter } from 'events';
+
+/**
+ * Meta Data containing information about the uploaded file.
+ * @beta
+ */
+export interface FileMetaData {
+  /**
+   * File name to be displayed.
+   */
+  name: string;
+  /**
+   * Extension is used for rendering the file icon.
+   * An unknown extension will be rendered as a generic icon.
+   * Example: `jpeg`
+   */
+  extension: string;
+  /**
+   * Download URL for the file.
+   */
+  url: string;
+}
+
+/**
+ * @beta
+ */
+export const UPLOAD_PROGRESSED_EVENT = 'uploadProgressed';
+/**
+ * @beta
+ */
+export const UPLOAD_COMPLETED_EVENT = 'uploadCompleted';
+/**
+ * @beta
+ */
+export const UPLOAD_FAILED_EVENT = 'uploadFailed';
+/**
+ * @beta
+ */
+export const UPLOAD_CANCELLED_EVENT = 'uploadCancelled';
+
+/**
+ * Events emitted by the UploadedFile class.
+ * @beta
+ */
+export type UploadedFileEvents =
```

Comment:
Why is this not simply `UploadedFileEvent = 'uploadFailed' | 'uploadCancelled'` etc.

Actually, I'd shorten the values to `UploadedFileEvent` = 'success' | 'cancelled' | 'failed'` etc.

Reply:
To avoid bugs caused due to spelling mistakes.
Using constants eliminates that.
I am against using generic event names. The code becomes confusing when other objects subscribe to these events and you have no idea what a `success` or a `failed` means in that context.